src/panel_material_ui/plotly.py

```diff
@@ -0,0 +1,492 @@
+
+    >>> pmui.plotly.register_pmui_templates(
+    ...     primary_color="#1976d2",
+    ...     margin={"l": 80, "r": 80, "t": 100, "b": 80}
+    ... )
+    """
+    _check_plotly_available()
+
+    # Determine colors for each theme
+    light_color = primary_color
+    dark_color = primary_color_dark if primary_color_dark is not None else primary_color
+
+    # Create both light and dark templates
+    light_template = create_pmui_template("light", primary_color=light_color, **kwargs)
+    dark_template = create_pmui_template("dark", primary_color=dark_color, **kwargs)
+
+    # Register with Plotly
+    pio.templates["pmui_light"] = light_template
+    pio.templates["pmui_dark"] = dark_template
+
+    # Validate that the specified default template exists
+    if default not in pio.templates:
+        raise ValueError(
+            f"Default template '{default}' not found. "
+            f"Available templates: {list(pio.templates.keys())}"
+        )
+
+    # Set specified template as default
+    pio.templates.default = default
+
+def update_template(
+    figure: Any,
+    dark_theme: bool = False
+) -> Any:
+    """Update a Plotly figure's template based on theme mode.
+
+    This function dynamically updates an existing Plotly figure to use
+    the appropriate Panel Material UI template based on the theme mode.
+    Useful for reactive theme switching in Panel applications.
+
+    Parameters
+    ----------
+    figure : plotly.graph_objects.Figure
+        The Plotly figure to update with the new template.
+    dark_theme : bool, default False
+        Whether to use the dark theme. If True, uses "pmui_dark" template,
+        otherwise uses "pmui_light" template.
+
+    Returns
+    -------
+    plotly.graph_objects.Figure
+        The updated figure with the new template applied.
+
+    Examples
+    --------
+    Update a figure to use dark theme:
+
+    >>> import plotly.express as px
+    >>> import panel_material_ui as pmui
+    >>>
+    >>> fig = px.scatter(df, x="x", y="y")
+    >>> dark_fig = pmui.plotly.update_template(fig, dark_theme=True)
+
+    Use with Panel's reactive system:
+
+    >>> import panel as pn
+    >>> toggle = pmui.ThemeToggle()
+    >>> pn.pane.Plotly(pn.bind(pmui.plotly.update_template, fig, toggle))
+
+    Notes
+    -----
+    This function assumes that Panel Material UI templates have been
+    registered. If templates are not found, an error will be raised.
+    """
+    _check_plotly_available()
+
+    template_name = "pmui_dark" if dark_theme else "pmui_light"
+
+    # Check if the template exists before applying
+    if template_name not in pio.templates:
+        raise ValueError(
+            f"Template '{template_name}' not found. "
+            "Make sure to call pmui.plotly.register_pmui_templates() first."
+        )
+
+    figure.update_layout(template=template_name)
+    return figure
+
+
+# Auto-register templates for convenience when plotly is available
+if PLOTLY_AVAILABLE:
```

Comment:
I decided to autoregister for ease of use in line with what Vizro does. But I don't how much it slows down the package import of if it has other consequences.

Otherwise we need to explain to users in plotly.md how to manually register templates - could also be an option.

But then the functions in plotly.py needs to be cached.